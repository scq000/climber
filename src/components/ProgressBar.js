import React from 'react';
import {StyleSheet, View} from 'react-native';
import {AnimatedCircularProgress} from 'react-native-circular-progress';
import Text from './Text';

const ProgressBar = ({percentage, amount}) => {
  return (
    <AnimatedCircularProgress
      size={300}
      width={30}
      fill={percentage}
      rotation={270}
      backgroundColor="#d8d8d8"
      backgroundWidth={30}
      tintColor="#ff0000"
      lineCap="round">
      {() => (
        <View style={styles.container}>
          <Text style={styles.steps}>Steps</Text>
          <Text style={styles.amount}>{amount}</Text>
          <Text style={styles.day}>Today</Text>
        </View>
      )}
    </AnimatedCircularProgress>
  );
};

const styles = StyleSheet.create({
  container: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  steps: {
    color: '#FFFFFF',
    fontSize: 35,
  },
  amount: {
    color: '#FFFFFF',
    fontSize: 50,
    fontWeight: 'bold',
  },
  day: {
    color: '#FFFFFF',
    fontSize: 16,
  },
});

export default ProgressBar;
