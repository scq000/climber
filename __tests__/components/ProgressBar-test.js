import 'react-native';
import React from 'react';
import {create} from 'react-test-renderer';
import ProgressBar from '../../src/components/ProgressBar';

describe('ProgressBar', () => {
  it('should render correctly', function () {
    const snap = create(<ProgressBar percentage={10} amount={123} />).toJSON();
    expect(snap).toMatchSnapshot();
  });
});
