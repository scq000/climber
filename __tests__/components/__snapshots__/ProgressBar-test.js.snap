// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render correctly 1`] = `
<View>
  <RNSVGSvgView
    bbHeight={300}
    bbWidth={300}
    focusable={false}
    height={300}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 300,
          "width": 300,
        },
      ]
    }
    width={300}
  >
    <RNSVGGroup>
      <RNSVGGroup
        matrix={
          Array [
            -1.8369701987210297e-16,
            -1,
            1,
            -1.8369701987210297e-16,
            2.842170943040401e-14,
            300,
          ]
        }
      >
        <RNSVGPath
          d="M 149.91517700393416 15.000026647931008 A 135 135 0 1 0 150 15"
          fill={0}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeLinecap",
            ]
          }
          stroke={4292401368}
          strokeLinecap={1}
          strokeWidth={30}
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderRadius": 120,
        "height": 240,
        "justifyContent": "center",
        "left": 30,
        "overflow": "hidden",
        "position": "absolute",
        "top": 30,
        "width": 240,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Helvetica",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 35,
            },
          ]
        }
      >
        Steps
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Helvetica",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 50,
              "fontWeight": "bold",
            },
          ]
        }
      >
        123
      </Text>
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Helvetica",
            },
            Object {
              "color": "#FFFFFF",
              "fontSize": 16,
            },
          ]
        }
      >
        Today
      </Text>
    </View>
  </View>
</View>
`;
